### ISSUE TITLE
AdChoices Logo View Disappear Randomly in UICollectionCell.

### ISSUE THREAD

**Issue by chrisma0101** [googleads/googleads-mobile-ios-examples#66]
Hi,

   I came across a iOS bug about AdChoices logo view will disappear when using GADUnifiedNativeAdView inside UICollectionViewCell. Because AdChoices view is **automatically** inserted by GoogleMobileAds SDK, I cannot find a way manually fixing this bug.
   Here is the scenario:
       1. Create UICollectionViewCell and create GADUnifiedNativeAdView as the first subview in .xib file;
       2. Then follow your documentation to add subviews under GADUnifiedNativeAdView, and then link all subviews to IBOutlet in GADUnifiedNativeAdView.h file.
       3. Create adLoader:
      `adLoader = GADAdLoader(
            adUnitID: "ca-app-pub-3940256099942544/3986624511",
            rootViewController: self,
            adTypes: [.unifiedNative],
            options: nil
        )`
        `adLoader?.delegate = self`
        `adLoader?.load(DFPRequest())`
       4. Once load finished, in the delegate method, call reloadData.
      `func adLoader(_ adLoader: GADAdLoader, didReceive nativeAd: GADUnifiedNativeAd) {`
              `ads.append(nativeAd)`
              `self.collectionView.reloadData()}`
      5. Add a "load" UIButton on screen, then whenever the "load" button gets tapped, it will call `adLoader?.load(DFPRequest())`
      6. You will notice the ad will be added into UICollectionView one-by-one. When you add more and more, and scrolling down, you will notice, in some cell, the AdChoices logo view will disappear randomly.

Could you please take a look at this bug? Please let me know if you need more information.

Thanks,
Chris

**Comment by stowy** [comment]
Hi, would you be able to attach your sample project that reproduces the issue?

**Comment by chrisma0101** [comment]
Hi Samuel,
    
    Thanks a lot for following up this issue. Here is the sample project for testing purpose (https://www.dropbox.com/s/phxodbqmch970gf/VintedSample%202.zip?dl=0). Running it on simulator or real device and keep clicking "Refresh" Button, then scrolling up and down, you will notice the AdChoices logo in some cells has disappeared or appeared randomly.

Best regards,
Chris

**Comment by stowy** [comment]
Hi @chrisma0101 - I don't have permissions to access the link you provided. Can you make it public?

**Comment by chrisma0101** [comment]
Hi Samuel,
    Sorry about not making it public. Please try this link for sample project. https://drive.google.com/file/d/1xS8xEF_2gbmm97-aUjU9uUNLBw0PwGUu/view?usp=sharing

Thanks,
Chris

**Comment by stowy** [comment]
thanks, I was able to reproduce. One thing I would suggest is to set the contentAdView.nativeAd property to nil in prepare for reuse, as follows:

```
class NativeContentAdCell: UICollectionViewCell {
    
    @IBOutlet weak var containerView: UIView!
    @IBOutlet weak var contentAdView: GADUnifiedNativeAdView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }
    
    override func prepareForReuse() {
        contentAdView.nativeAd = nil
        super.prepareForReuse()
    }
...
```

However, this doesn't seem to resolve the issue. I will ask the engineering team to take a look.

**Comment by stowy** [comment]
One fix however is to not reload the entire collection view when adding a new ad. When the new ad is inserted like this:

```
extension ViewController: GADUnifiedNativeAdLoaderDelegate {
    func adLoader(_ adLoader: GADAdLoader, didReceive nativeAd: GADUnifiedNativeAd) {
        ads.append(nativeAd)
        self.collectionView.insertItems(at: [IndexPath(item: ads.count-1, section: 0)])
    }
}
```

Then the behavior does not occur.  I suspect the issue is related to having a certain number of ads on screen at one time. This should not be an issue, but if there are items interspersed with the ads or you do not reload native ad cells, it's unlikely to occur.

**Comment by stowy** [comment]
Hi Chris, can you try another workaround? This is not optimal, but instead of caching ads, create a `GADUnifiedNativeAdView` for each new ad and cache those instead, and then in your cell, instead of updating the ad view with a new ad, just remove the existing ad view and add the new one.

We are looking at a fix on our end, however this workaround may work for now.

**Comment by stowy** [comment]
This will be fixed in the next release due out this August.

**Comment by stowy** [comment]
Fixed in [version 7.32.0](https://developers.google.com/admob/ios/rel-notes) of the SDK.

**Comment by abinhho** [comment]
Same issue using SwiftUI, LazyVStack.
![Screenshot 2024-05-15 at 16 53 20](https://github.com/googleads/googleads-mobile-ios-examples/assets/9253378/d04b8e80-8e18-4064-9bb9-2d12daf7f3fc)

It's happening randomly.
